Problem1.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Definition for a binary tree node.
+ * public class TreeNode {
+ *     int val;
+ *     TreeNode left;
+ *     TreeNode right;
+ *     TreeNode(int x) { val = x; }
+ * }
+ */
+
+/**
+LeetCode Submitted : YES
+Space Complexity : O(1)
+Time Complexity : O(N)
```

Comment:
are you neglecting the queue used for BFS?

Reply:
No, this complexity is incorrect. 
Let me fix it.

Reply:
ok